tideways/provision.sh

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+# Tideways with XHgui
+DIR=`dirname $0`
+
+install_tideways() {
+    # Tideways is only for php =>7.0
+    echo "Installing/update Tideways to PHP 7.0, 7.1, 7.2"
+    git clone "https://github.com/tideways/php-xhprof-extension" "/var/local/tideways-php7.2"
+    cp -r /var/local/tideways-php7.2 /var/local/tideways-php7.0
+    cp -r /var/local/tideways-php7.2 /var/local/tideways-php7.1
+    for version in 7.0 7.1 7.2
+        do
+        cd "/var/local/tideways-php${version}"
+        update-alternatives --set php /usr/bin/php$version
+        update-alternatives --set php-config /usr/bin/php-config$version
+        update-alternatives --set phpize /usr/bin/phpize$version
+        phpize$version
+        ./configure --enable-tideways-xhprof --with-php-config=php-config$version
+        make
+        make install
+    done
+}
+
+restart_php() {
+    if [[ -d "/etc/php/7.0/" ]]; then
+        service php7.0-fpm restart
+    fi
+    if [[ -d "/etc/php/7.0/" ]]; then
+        service php7.1-fpm restart
+    fi
+    if [[ -d "/etc/php/7.0/" ]]; then
+        service php7.2-fpm restart
+    fi
+}
+
+install_mongodb() {
+    apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 0C49F3730359A14518585931BC711F9BA15703C6
```

Comment:
This sometimes fails, it would be better to grab the key in advance and store it in the repo